float32 rpm
float32 angle
